src/private/files.ts

```diff
@@ -0,0 +1,160 @@
+import { sendMessageToParent } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { FileOpenPreference, SdkError } from '../public';
+
+/**
+ * Namespace to interact with the files specific part of the SDK.
+ *
+ * @private
+ * Hide from docs
+ */
+export namespace files {
+  /**
+   * @private
+   * Hide from docs
+   *
+   * Cloud storage providers registered with Microsoft Teams
+   */
+  export enum CloudStorageProviderCode {
+    Dropbox = 'DROPBOX',
+    Box = 'BOX',
+    Sharefile = 'SHAREFILE',
+    GoogleDrive = 'GOOGLEDRIVE',
+    Egnyte = 'EGNYTE',
+  }
+
+  /**
+   * @private
+   * Hide from docs
+   *
+   * Cloud storage provider integration type
+   */
+  export enum CloudStorageProviderType {
+    Sharepoint = 0,
+    WopiIntegration = 1,
+    Google = 2,
+  }
+
+  /**
+   * @private
+   * Hide from docs
+   *
+   * Cloud storage folder interface
+   */
+  export interface CloudStorageFolder {
+    /**
+     * ID of the cloud storage folder
+     */
+    id: string;
+    /**
+     * Display Name/Title of the cloud storage folder
+     */
+    title: string;
+    /**
+     * ID of the cloud storage folder in the provider
+     */
+    folderId: string;
+    /**
+     * Type of the cloud storage folder provider integration
+     */
+    providerType: CloudStorageProviderType;
+    /**
+     * Code of the supported cloud storage folder provider
+     */
+    providerCode: CloudStorageProviderCode;
+    /**
+     * Display name of the owner of the cloud storage folder provider
+     */
+    ownerDisplayName: string;
+    /**
+     * Sharepoint specific siteURL of the folder
+     */
+    siteUrl?: string;
+    /**
+     * Sharepoint specific serverRelativeUrl of the folder
+     */
+    serverRelativeUrl?: string;
+    /**
+     * Sharepoint specific libraryType of the folder
+     */
+    libraryType?: string;
+    /**
+     * Sharepoint specific accessType of the folder
+     */
+    accessType?: string;
+  }
+
+  /**
+   * @private
+   * Hide from docs
+   *
+   * Gets a list of cloud storage folders added to the channel
+   * @param channelId ID of the channel whose cloud storage folders should be retrieved
+   * @param callback Callback that will be triggered post folders load
+   */
+  export function getCloudStorageFolders(
+    channelId: string,
+    callback: (error: SdkError, folders: CloudStorageFolder[]) => void,
+  ): void {
+    if (!channelId || channelId.length == 0) {
+      throw new Error('[files.getCloudStorageFolders] channelId name cannot be null or empty');
+    }
+    if (!callback) {
+      throw new Error('[files.getCloudStorageFolders] Callback cannot be null');
+    }
+
+    ensureInitialized();
```

Comment:
Please changes this to `ensureInitialized(FrameContexts.content)` since this API should never be called from any other frame context.